app/models/like.rb

```diff
@@ -1,4 +1,9 @@
 class Like < ApplicationRecord
-  belongs_to :user
-  belongs_to :post
+  belongs_to :user, optional: true
+  belongs_to :post, optional: true, foreign_key: :posts_id
```

Comment:
- [ ] The same thing as the previous comment. Kindly specify the user as an `author` and use the `class_name` method for making it valid.
- [ ] Also, the foreign_key in the post association isn't necessary so please remove it